docs/src/reference/types.md

```diff
@@ -174,6 +174,145 @@ Produces the negative of the color.
+  should be displayed as a number or a word. Unfortunately, when choosing the 
+  word representation, it can currently only display the English version. In 
+  the future, it is planned to support localization.
+* `day`: Displays the day of the datetime.
+  * `padding`: Can be either `zero`, `space` or `none`. Specifies how the day 
+  is padded.
+* `week_number`: Displays the week number of the datetime.
+  * `padding`: Can be either `zero`, `space` or `none`. Specifies how the week 
+  number is padded.
+  * `repr`: Can be either `ISO`, `sunday` or `monday`. In the case of `ISO`, 
+  week numbers are between 1 and 53, while the other ones are between 0 and 53.
+* `weekday`: Displays the weekday of the date.
+  * `repr` Can be either `long`, `short`, `sunday` or `monday`. In the case of 
+  `long` and `short`, the corresponding English name will be displayed (same as 
+  for the month, other languages are currently not supported). In the case of 
+  `sunday` and `monday`, the numerical value will be displayed (assuming Sunday 
+  and Monday as the first day of the week, respectively).
+  * `one_indexed`: Can be either `true` or `false`. Defines whether the 
+  numerical representation of the week starts with 0 or 1.
+* `hour`: Displays the hour of the date.
+  * `padding`: Can be either `zero`, `space` or `none`. Specifies how the hour
+  is padded.
+  * `repr`: Can be either `24` or `12`. Changes whether the hour is displayed in
+  the 24-hour or 12-hour format.
+* `period`: The AM/PM part of the hour
+  * `case`: Can be `lower` to display it in lower case and `upper` to display it
+    in upper case.
+* `minute`: Displays the minute of the date.
+  * `padding`: Can be either `zero`, `space` or `none`. Specifies how the minute
+    is padded.
+* `second`: Displays the second of the date.
+  * `padding`: Can be either `zero`, `space` or `none`. Specifies how the second
+    is padded.
+
+Keep in mind that not always all components can be used. For example, if
+you create a new datetime with `#datetime(year: 2023, month: 10, day: 13)`, it
+will be stored as a plain date internally, meaning that you cannot use 
+components such as `hour` or `minute`, which would only work on datetimes 
+that have a specified time.
+
+- pattern: string (positional)
+  The format used to display the datetime.
+- returns: string
+
+```example
+#let date = datetime(year: 2020, month: 10, day: 4)
+
+Default representation: #date.display() \
+Custom representation: #date.display("Last two digits of year: [year padding:zero repr:last_two] Month: [month padding:zero], Weekday: [weekday repr:monday one_indexed:true] ([weekday repr:long]), Day: [day padding:none]")
+
+#let time = datetime(hour: 18, minute: 2, second: 23)
+
+Default representation: #time.display() \
+Custom representation: #time.display("Hour: [hour repr:12] ([period]), Minute: [minute padding:none], Second: [second]")
+```
+
+### year()
+
```

Comment:
```suggestion
```
This is of course a matter of taste, but I would prefer no blank lines after headings. :)